services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -0,0 +1,147 @@
+        kinds=["split-first-rows-from-streaming", "split-first-rows-from-parquet"],
+        dataset=dataset,
+        config=config,
+        split=split,
+    )
+    try:
+        first_rows = upstream_response.response["content"]
+        features = first_rows["features"]
+    except KeyError as e:
+        raise PreviousStepFormatError("Previous step did not return the expected content.", e) from e
+
+    # look for string columns using the first rows
+    string_columns = [
+        feature["name"]
+        for feature in features
+        if "dtype" in feature["type"]
+        and "_type" in feature["type"]
+        and feature["type"]["dtype"] == STRING_FEATURE_DTYPE
+        and feature["type"]["_type"] == VALUE_FEATURE_TYPE
+    ]
+
+    if not string_columns:
+        raise NoIndexableColumnsError("No string columns available to index.")
+
+    # look for image, audio and binary columns, if present, raise exeception do not supported yet and index everything
+    if any(
+        feature["name"]
+        for feature in features
+        if "_type" in feature["type"] and feature["type"]["_type"] in UNSUPPORTED_FEATURES_MAGIC_STRINGS
+    ):
+        raise UnsupportedIndexableColumnsError("Unsupported feature types for indexing.")
+
+    # get list of parquet urls
+    config_parquet = get_previous_step_or_raise(kinds=["config-parquet"], dataset=dataset, config=config)
+    try:
+        parquet_files = config_parquet.response["content"]["parquet_files"]
+        parquet_urls = [content["url"] for content in parquet_files if content["split"] == split]
+
+        if not parquet_urls:
+            raise ParquetResponseEmptyError("No parquet files found.")
+    except Exception as e:
+        raise PreviousStepFormatError("Previous step did not return the expected content.") from e
+
+    # create duckdb index location
+    split_path, dir_path = create_index_dir_split(
+        dataset=dataset, config=config, split=split, index_directory=duckdb_index_directory
+    )
+    duck_db_name = f"{split_path}/{DUCKDB_DEFAULT_DB_NAME}"
+    db_location = dir_path / DUCKDB_DEFAULT_DB_NAME
+
+    # configure duckdb extensions
+    duckdb.execute(INSTALL_EXTENSION_COMMAND.format(extension="httpfs"))
+    duckdb.execute(LOAD_EXTENSION_COMMAND.format(extension="httpfs"))
+    duckdb.execute(INSTALL_EXTENSION_COMMAND.format(extension="fts"))
+    duckdb.execute(LOAD_EXTENSION_COMMAND.format(extension="fts"))
+
+    # index
+    con = duckdb.connect(str(db_location))
+    con.sql(CREATE_SEQUENCE_COMMAND)
+    con.sql(f"{CREATE_TABLE_COMMAND} read_parquet({parquet_urls});")
+    con.sql(DROP_INDEX_COMMAND)
+
+    # TODO: by default, 'porter' stemmer is being used, use a specific one by dataset language in the future
+    # see https://duckdb.org/docs/extensions/full_text_search.html for more deails about 'stemmer' parameter
+    con.sql(CREATE_INDEX_COMMAND)
+
+    return IndexRowsResponse(duckdb_db_name=duck_db_name)
```

Comment:
duck_db_name => duckdb_index_filename?

Reply:
I changed the response structure to:
```
    dataset: str
    config: str
    split: str
    url: str
    filename: str
    size: int
```
I think it makes more sense to have similar format as we have on /parquet